### PR TITLE
[5.0] Extracted Contracts

### PR DESCRIPTION
**Pull request by ibrasho**
Extracted Database, View, Validation, Translation and Queue contracts.

Fixed some typos along the way

Only ones remaining are:
- `Illuminate\Routing\Annotations\EndpointInterface`
- `Illuminate\Routing\Matching\ValidatorInterface`

### CODE REVIEW

**Comment by taylorotwell** [comment]
One thing I noticed with this is it makes the contracts type hint the Symfony HttpFoundation which we can't really do because that would put a dependency on HttpFoundation from contracts, which we want to have zero dependencies. :cry:

**Comment by ibrasho** [comment]
Only `Illuminate\Contracts\Session\Store` does that, and we can keep it in `Illuminate\Session`.

What do you think?

**Comment by GrahamCampbell** [comment]
No, that won't work. We can add typehints to implementations of interfaces, PHP will fatal error if you try to do that.

**Comment by ibrasho** [comment]
@taylorotwell I just updated the files to return the `Illuminate\Contracts\Session\Store` interface to the `Illuminate\Session` namespace, therefore `Illuminate\Contracts` doesn't depend on `Symfony/HttpFoundation` anymore.

@GrahamCampbell could you please ensure I didn't break anything in terms of style? ; )

The conflict is due to the fact that around 150 commit happened after I made this. Mostly docblocks and imports. I'll work on getting it fixed.

**Comment by GrahamCampbell** [comment]
Ping @ibrasho.

**Comment by ibrasho** [comment]
I'm thinking of submitting this again in splits instead of all-at-once.
This should give better visibility to the changes.

Any changes that are still required that I'm missing?

**Comment by GrahamCampbell** [comment]
Can you just update this pull please. I want to do the psr-2 suff after this has been either merged or rejected.

**Comment by ibrasho** [comment]
I work on fixing the merge conflicts now. Anything else?

**Comment by GrahamCampbell** [comment]
Nothing I can see atm.

**Comment by ibrasho** [comment]
Done.

One last CS check?

File: src/Illuminate/Foundation/Application.php [hunk 1]
```diff
@@ -820,7 +820,7 @@ public function registerCoreContainerAliases()
 			'request'        => 'Illuminate\Http\Request',
 			'router'         => ['Illuminate\Routing\Router', 'Illuminate\Contracts\Routing\Registrar'],
 			'session'        => 'Illuminate\Session\SessionManager',
-			'session.store'  => ['Illuminate\Session\Store', 'Symfony\Component\HttpFoundation\Session\SessionInterface'],
+			'session.store'  => ['Illuminate\Session\StoreInterface', 'Symfony\Component\HttpFoundation\Session\SessionInterface'],
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Application.php]: Shouldn't this be a contract?

File: src/Illuminate/Http/RedirectResponse.php [hunk 2]
```diff
@@ -21,7 +21,7 @@ class RedirectResponse extends BaseRedirectResponse {
 	/**
 	 * The session store implementation.
 	 *
-	 * @var \Illuminate\Session\Store
+	 * @var \Illuminate\Session\StoreInterface
```
Comment (GrahamCampbell) [inline src/Illuminate/Http/RedirectResponse.php]: Shouldn't this be a contract?

**Comment by GrahamCampbell** [comment]
Ping @ibrasho

File: src/Illuminate/Http/RedirectResponse.php (hunk 2 above)
Comment (ibrasho) [inline src/Illuminate/Http/RedirectResponse.php]: I revert this due the concern @taylorotwell raised regarding the dependency on Symfony/HttpFoundation.

**Comment by GrahamCampbell** [comment]
Really sorry. This needs rebasing again. Should be quite quick this time around.

**Comment by ibrasho** [comment]
Rebased again. :-)

**Comment by ibrasho** [comment]
@taylorotwell What do you think? `laravel/contracts` does not depend on `symfony/httpfoundation` now.

**Comment by ibrasho** [comment]
More changes required now after @taylorotwell extracted UserProviderInterface and StoreInterface

**Comment by GrahamCampbell** [comment]
I had a chat to Taylor about this a few days back, and he actually isn't too keen on this. Is there any chance you could send the minor fixes you made that were unrelated to the namespaces in another pull for now?

**Pull request closed without merging**